WearComplicationProvidersTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/IconProviderService.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.wear.wearcomplicationproviderstestsuite
+
+import android.content.ComponentName
+import android.graphics.drawable.Icon
+import android.support.wearable.complications.ComplicationData
+import android.support.wearable.complications.ComplicationManager
+
+/**
+ * A complication provider that supports only [ComplicationData.TYPE_ICON] and cycles through
+ * a few different icons on each tap.
+ */
+class IconProviderService : SuspendingComplicationProviderService() {
+    override suspend fun onComplicationUpdateImpl(complicationId: Int, type: Int, manager: ComplicationManager) {
+        if (type != ComplicationData.TYPE_ICON) {
+            manager.noUpdateRequired(complicationId)
+            return
+        }
+        val args = ComplicationToggleArgs(
+            providerComponent = ComponentName(this, javaClass),
+            complicationId = complicationId,
+        )
+        val complicationTogglePendingIntent =
+            ComplicationToggleReceiver.getComplicationToggleIntent(
+                context = this,
+                args = args,
+            )
+        val state = args.getState(this@IconProviderService)
```

Comment:
I would add a comment here (and the other Services where you do this about what is going on... It might not be clear right away why you creating a version of the sample that enables coroutines. Maybe something like this?

// Retrieves state from DataStore (suspend method).

Reply:
Nit on the review comment: "suspend **function**" is more accurate than "suspend _method_"

Reply:
Added!

(It's a shame that the "suspension point" gutter icons only show up in the IDE, since they are really helpful to see)